Compiler/BackEnd/Tearing.mo

```diff
@@ -4028,5 +3992,16 @@ algorithm
 
 end recursiveTearingReplace;
 
+protected function getUnassigned
+  input array<Integer> ass;
+  output list<Integer> unassigned={};
+algorithm
+  for i in 1:arrayLength(ass) loop
+    if Dangerous.arrayGetNoBoundsChecking(ass, i)==-1 then
+      unassigned := i::unassigned;
+    end if;
+  end for;
+end getUnassigned;
+
```

Comment:
is there a difference between this and
unassigned := list(i for i guard Dangerous.arrayGetNoBoundsChecking(ass, i)==-1 in arrayLength(ass):-1:1) 
for the compiler?